docs/FlaskSQLAlchemy.md

```diff
@@ -0,0 +1,419 @@
+## Flask-SQLAlchemy
+
+Flask-SQLAlchemy is an extension for [Flask](https://flask.palletsprojects.com/en/1.1.x/) that adds support for [SQLAlchemy](https://www.sqlalchemy.org/) to your application.
+
+## SQLAlchemy
+
+SQLAlchemy is the Python SQL toolkit and Object Relational Mapper that gives application developers the full power and flexibility of SQL.
+Or simply it can be understood as a database toolkit for python. It represents end-to-end system for working with python DBAPI (Database API), relational databases, & the SQL language.It provides helpers, tools & components to assist with database development at every level.
+
+Two main components of SQLAlchemy:
+* SQLAlchemy Core
+* SQLAlchemy ORM
+
+### SQLAlchemy Core
+
+* **Engine** - a registry (object in a computer or an application that stores some common repository of information about something) which provides connectivity to a particular database server. In this case the engine stores information about the database it is connected to.
+* **Dialect** - interprets generic SQL and database commands in terms of a specific DBAPI and database backend.
+* **Connection Pool** - holds a collection of database connections in memory for fast re-use.
+* **SQL Expression Language** - Allows SQL satements to be written using Python expressions.
+* **Schema/Types** - Uses Python objects to represent tables, columns, and datatypes.
+
+### SQLAlchemy ORM
+
+SQLAlchemy is most famous for its object-relational mapper (ORM) or Object-relational mapping is a technique for storing, retrieving, updating, and deleting from an object-oriented program in a relational database.
+
+* The ORM allows construction of python object which can be mapped to relational database tables
+* Transparently persists objects(like the state of a python object) into their corresponding database tables using the unit of work pattern.
+* Provides a query system which loads objects and attributes using SQL generated from mappings.
+* Builds on top of the core- uses the Core to generate SQL and talk to the database.
+
+**Note:** For the complete guide, checkout the API documentation on the [SQLAlchemy](https://flask-sqlalchemy.palletsprojects.com/en/2.x/api/#flask_sqlalchemy.SQLAlchemy) class.
+
+### Requirements
+
+| Our Version             | Python                  | Flask                   | SQLAlchemy              |
+|-------------------------|-------------------------|-------------------------|-------------------------|
+| 2.x                     | 2.7, 3.4+               | 0.12+                   | 0.8+ or 1.0.10+ w/      |
+|                         |                         |                         | Python 3.7              |
+| 3.0+ (in dev)           | 2.7, 3.5+               | 1.0+                    | 1.0+                    |
```

Comment:
Why you wrote Python 3.7 in SQLAlchemy column?

Reply:
https://flask-sqlalchemy.palletsprojects.com/en/2.x/
have a look here once I don't exactly know what this means

Reply:
I guess it means that SQLAlchemy of version 1.0.10+ is compatible with Python 3.7

Reply:
But Python versions are written in another column which are compatible with it. You should look into it once again

Reply:
Did u see the link?

Reply:
Ok I will look into it again

Reply:
I think it is just in the SQLAlchemy column because Python 3.7 is compatible with SQLAlchemy 1.0.10+ 
I will remove it. What do u think?